broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java

```diff
@@ -140,4 +149,74 @@ public ActorFuture<Void> resumeProcessing() {
         });
     return completed;
   }
+
+  @Override
+  public ActorFuture<Void> banInstance(final long processInstanceKey) {
+    final ActorFuture<Void> future = concurrencyControl.createFuture();
+    concurrencyControl.run(
+        () -> {
+          try {
+
+            final LogStream logStream = adminControl.getLogStream();
+            logStream
+                .newLogStreamWriter()
+                .onComplete(
+                    (writer, error) -> {
+                      if (error != null) {
+                        LOG.error(
+                            "Could not retrieve writer to write error record for process instance.",
+                            error);
+                        future.completeExceptionally(error);
+                      } else {
+                        final var errorRecord = new ErrorRecord();
+                        errorRecord.initErrorRecord(
+                            new Exception("Instance was banned from outside."), -1);
+                        errorRecord.setProcessInstanceKey(processInstanceKey);
+
+                        final var recordMetadata =
+                            new RecordMetadata()
+                                .recordType(RecordType.EVENT)
```

Comment:
🔧 I'd love for us to introduce a special record type for this `CORRECTION` to separate it from regular events. Otherwise, the -1 source record position would be the only thing indicating that this is a correction. Which is okay, but less obvious.

Reply:
I guess would be fine to me. But needs a bit more changes involved like replay adjustments etc.